jenkins/Jenkinsfile.pull-request

```
@@ -32,10 +33,33 @@ pipeline {
           utils.addRemoteRef("${env.CHANGE_TARGET}")
           RSTUDIO_VERSION_FLOWER = readFile(file: 'version/RELEASE').replaceAll(" ", "-").toLowerCase().trim()
           IS_PRO = JOB_URL.contains('Pro')
+          BASE_IMAGE = "jenkins/ide:pro-jammy-x86_64-${RSTUDIO_VERSION_FLOWER}"
         }
       }
     }
 
+    stage('Snyk Code Scan') {
+      environment {
+        SNYK_TOKEN = credentials('snyk-auth-token')
+        TMP_SNYK_PATH = "/tmp/snyk-code-scan"
+      }
+      agent {
+        dockerfile {
+          filename 'docker/jenkins/Dockerfile.snyk'
+          registryCredentialsId 'ecr:us-east-1:aws-build-role'
+          registryUrl 'https://263245908434.dkr.ecr.us-east-1.amazonaws.com'
+          additionalBuildArgs "--build-arg BASE_IMAGE=${BASE_IMAGE}" + ' --build-arg GITHUB_LOGIN=${GITHUB_LOGIN}'
+          reuseNode true
+        }
+      }
+      steps {
+        sh "mkdir --parents ${TMP_SNYK_PATH}"
+        sh "cp --parents -r $(git diff --name-only ${env.CHANGE_TARGET}) ${TMP_SNYK_PATH}"
```

I'm seeing this error in the PR job:
```
2023-12-22 15:37:57  Obtained jenkins/Jenkinsfile.pull-request from d3bfbced81287930aaf32657ba3137d044e36065
2023-12-22 15:37:57  org.codehaus.groovy.control.MultipleCompilationErrorsException: startup failed:
2023-12-22 15:37:57  WorkflowScript: 57: illegal string body character after dollar sign;
2023-12-22 15:37:57     solution: either escape a literal dollar sign "\$5" or bracket the value expression "${5}" @ line 57, column 9.
2023-12-22 15:37:57             sh "cp --parents -r $(git diff --name-only ${env.CHANGE_TARGET}) ${TMP_SNYK_PATH}"
2023-12-22 15:37:57             ^
2023-12-22 15:37:57  
2023-12-22 15:37:57  1 error
```

I think you just need to escape the first dollar sign since you want Groovy to interpret that as a $ literal instead of string interpolation.

```suggestion
        sh "cp --parents -r \$(git diff --name-only ${env.CHANGE_TARGET}) ${TMP_SNYK_PATH}"
```

---

Good catch. Added the suggestion in.